panacea/grpc_client.go

```diff
@@ -16,7 +18,26 @@ type GrpcClient struct {
 
 func NewGrpcClient(conf *config.Config) (*GrpcClient, error) {
 	log.Infof("dialing to Panacea gRPC endpoint: %s", conf.Panacea.GRPCAddr)
-	conn, err := grpc.Dial(conf.Panacea.GRPCAddr, grpc.WithInsecure())
+
+	parsedUrl, err := url.Parse(conf.Panacea.GRPCAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse gRPC endpoint. please use absolute URL (scheme://host:port): %w", err)
+	}
+
+	var cred grpc.DialOption
+
+	if parsedUrl.Scheme == "https" {
+		tlsConfig, err := config.CreateTLSConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create TLS cofig: %w", err)
+		}
+		cred = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
```

Comment:
IIRC, we can just use a zero `tls.Config{}`.
```go
package main

import (
	"context"
	"crypto/tls"
	"fmt"
	"time"

	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials"
)

func main() {
	addr := "grpc.gopanacea.org:443"

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	creds := credentials.NewTLS(&tls.Config{})
	conn, err := grpc.DialContext(ctx, addr, grpc.WithTransportCredentials(creds))
	if err != nil {
		panic(err)
	}

	defer conn.Close()

	client := banktypes.NewQueryClient(conn)
	resp, err := client.TotalSupply(context.Background(), &banktypes.QueryTotalSupplyRequest{})
	if err != nil {
		panic(err)
	}

	fmt.Printf("%+v\n", resp)
}
```

Reply:
Oh, I didn't know that. I'll change it as you suggested. Thanks!